### ISSUE TITLE
Unable to install Docker Desktop 2.3.03 in Windows 10 Enterprise.  Installation fails.  

### ISSUE THREAD

**Issue by pldiguanaman** [docker/for-win#7407]
I cannot get the application to install into Windows 10 Enterprise.  Version 1903 (OS Build 18632.778)

### Expected behavior
I expect the application to install

### Actual behavior
Error during installation:
  at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   at CommunityInstaller.EnableFeaturesAction.<DoAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<ProcessAsync>d__24.MoveNext()

### Information

  - Windows Version:  Windows 10 Enterprise.  Version 1903 (OS Build 18632.778)
  - Docker Desktop Version: 2.3.03
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM:  Yes a VM 

### Steps to reproduce the behavior

  1. Simply try and install Docker and it fails.
  2. ...

**Comment by rmja** [comment]
I am seeing the same error. The following is the content of `install-log.txt`:

```
[08:43:19.348] [InstallHandler] No installation found
[08:43:19.351] [InstallHandler] Using embedded package
[08:43:19.605] [InstallWorkflow] Using package: res:CommunityInstaller.DockerDesktop.d4w
[08:43:19.607] [InstallWorkflow] Downloading
[08:43:20.757] [InstallWorkflow] Extracting manifest
[08:43:22.720] [InstallWorkflow] Checking prerequisites
[08:43:22.831] [InstallWorkflow] Unpacking artifacts
[08:44:57.255] [InstallWorkflow] Deploying component CommunityInstaller.CreateGroupAction
[08:45:04.128] [InstallWorkflow] Deploying component CommunityInstaller.AddToGroupAction
[08:45:10.968] [InstallWorkflow] Deploying component CommunityInstaller.EnableFeaturesAction
[08:45:11.117] [InstallWorkflow] Component CommunityInstaller.EnableFeaturesAction failed
Exception type: System.Runtime.InteropServices.COMException, Exception message: , StackTrace:
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   at CommunityInstaller.EnableFeaturesAction.<DoAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__29.MoveNext()
[08:45:11.119] [InstallWorkflow] Installation failed
Exception type: System.Runtime.InteropServices.COMException, Exception message: , StackTrace:
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   at CommunityInstaller.EnableFeaturesAction.<DoAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<ProcessAsync>d__24.MoveNext()
[08:45:11.122] [InstallWorkflow] Rollbacking component CommunityInstaller.AddToGroupAction
[08:45:11.124] [InstallWorkflow] Rollbacking component CommunityInstaller.CreateGroupAction
[08:45:11.126] [InstallWorkflow] Rollbacking component CommunityInstaller.UnpackArtifactsStep
[08:45:11.140] [FileSystem] Deleting C:\Program Files\Docker\Docker
[08:45:11.421] [FileSystem] Deleted C:\Program Files\Docker\Docker successfully
[08:45:11.422] [InstallWorkflow] Rollbacking component CommunityInstaller.ExtractManifestStep
[08:45:11.424] [InstallWorkflow] Rollbacking component CommunityInstaller.DownloadStep
```

**Comment by rmja** [comment]
It seems as if deleting everything from `AppData/Local/Docker` and `AppData/Roaming/Docker*` before trying the install does not resolve the problem... :(

**Comment by hsimah** [comment]
Still an issue with 2.3.0.4 today. Exact same error, I even tried clearing out app data before seeing this.

**Comment by Nuvelle** [comment]
Unable to install 2.3.0.4 (46911) with the same issue


```
[10:48:24.819] [InstallHandler] No installation found
[10:48:24.821] [InstallHandler] Using embedded package
[10:48:25.047] [InstallWorkflow] Using package: res:CommunityInstaller.DockerDesktop.d4w
[10:48:25.049] [InstallWorkflow] Downloading
[10:48:26.719] [InstallWorkflow] Extracting manifest
[10:48:36.932] [InstallWorkflow] Checking prerequisites
[10:48:46.968] [InstallWorkflow] Unpacking artifacts
[10:49:49.983] [InstallWorkflow] Deploying component CommunityInstaller.CreateGroupAction
[10:49:59.022] [InstallWorkflow] Deploying component CommunityInstaller.AddToGroupAction
[10:50:08.036] [InstallWorkflow] Deploying component CommunityInstaller.EnableFeaturesAction
[10:50:18.075] [InstallWorkflow] Component CommunityInstaller.EnableFeaturesAction failed
Exception type: System.BadImageFormatException, Exception message: , StackTrace:
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   at CommunityInstaller.EnableFeaturesAction.<DoAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__34.MoveNext()
[10:50:18.076] [InstallWorkflow] Installation failed
Exception type: System.BadImageFormatException, Exception message: , StackTrace:
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Management.ManagementObjectCollection.ManagementObjectEnumerator.MoveNext()
   at System.Linq.Enumerable.<CastIterator>d__97`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at CommunityInstaller.EnableFeaturesAction.GetFeaturesToEnable()
   at CommunityInstaller.EnableFeaturesAction.<DoAsync>d__37.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__34.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__29.MoveNext()
[10:50:18.078] [InstallWorkflow] Rollbacking component CommunityInstaller.AddToGroupAction
[10:50:18.079] [InstallWorkflow] Rollbacking component CommunityInstaller.CreateGroupAction
[10:50:18.080] [InstallWorkflow] Rollbacking component CommunityInstaller.UnpackArtifactsStep
[10:50:18.087] [FileSystem] Deleting C:\Program Files\Docker\Docker
[10:50:18.311] [FileSystem] Deleted C:\Program Files\Docker\Docker successfully
[10:50:18.312] [InstallWorkflow] Rollbacking component CommunityInstaller.ExtractManifestStep
[10:50:18.313] [InstallWorkflow] Rollbacking component CommunityInstaller.DownloadStep
```

**Comment by RNabla** [comment]
Was getting the same error. In my case leaving **Enable Hyper-V feature'** unchecked resolved the issue and installion succeeded.
I am a Docker newbie so I don't know if there any impact of not having hyper-v enabled.

OS Name:                   Microsoft Windows 10 Education N
OS Version:                10.0.17763 N/A Build 1
Docker installer: 2.3.0.4 (46911)

**Comment by hsimah** [comment]
I stupidly allowed an update to install, and this happened again.

What I did today was get the latest stable from the website. Deleted all Docker folders and files from app data local and roaming. Did a clean install without checking the WSL2 option. Restarted the PC and ran Docker Desktop. It failed to start so I used the option to reset to defaults. This restarted Docker Desktop and it is now working. I had to rebuild all my containers but hopefully this helps.

**Comment by RNabla** [comment]
> 
> 
> Was getting the same error. In my case leaving **Enable Hyper-V feature'** unchecked resolved the issue and installion succeeded.
> I am a Docker newbie so I don't know if disabling this option has some impact on Docker.
> 
> OS Name: Microsoft Windows 10 Education N
> OS Version: 10.0.17763 N/A Build 1
> Docker installer: 2.3.0.4 (46911)

Atleast for me unchecking 'Hyper-V' resulted in getting runtime error when starting Docker Desktop. 
I've checked 'Turn Windows features on and off' and I was having Hyper-V disabled. I've tried to enable that manually but I was getting some 0x**** error from Windows. It seems error that happened during enabling Hyper-V was related to some pending updates from Windows Update. Also in my case I've had problems with installing latest updates (installing windows updates failed with another 0x****). After spending some time I was unable to workaround failing windows updates. I've tried to use windows recovery tool but it was resulting in another 0x****. 
At the end of the day I've installed new windows and was able to successfully install Docker 2.3.0.4 (46911) on
OS Name:                   Microsoft Windows 10 Education N
OS Version:                10.0.19041 N/A Build 19041

So if any one is having problems I'd check for pending windows updates before installing Docker and/or enabling Hyper-V manually before installing.
